### PR TITLE
task/DEP-3118: Add HPA to generic3 chart

### PR DESCRIPTION
**Pull request by wombolo**
- Enable Horizontal pod autoscaling on generic3 Helm chart
- Enable pre-commit helper to run `helm template` and package charts

### CODE REVIEW

File: src/generic3/templates/horizontal-pod-autoscaler.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    app: {{ template "generic.name" . }}
+    chart: {{ template "generic.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: {{ .Values.resourceType }}
+    name: {{ template "generic.fullname" . }}
+  minReplicas: {{ .Values.replicaCount }}
```
Comment (femilofin) [inline src/generic3/templates/horizontal-pod-autoscaler.yaml:15]: make this:
app:
  replica:
    enabled: false
    min:
    max:
    utilization:
      cpu: 70
      memory: 70
 or something similar

File: src/generic3/templates/horizontal-pod-autoscaler.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  maxReplicas: {{ .Values.maxReplicaCount }}
```
Comment (femilofin) [inline src/generic3/templates/horizontal-pod-autoscaler.yaml:16]: look at previous comment

File: src/generic3/templates/horizontal-pod-autoscaler.yaml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  metrics:
+    - type: Resource
+      resource:
+        name: cpu
+        targetAverageUtilization: 70
```
Comment (femilofin) [inline src/generic3/templates/horizontal-pod-autoscaler.yaml:21]: this too

File: src/generic3/templates/horizontal-pod-autoscaler.yaml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    - type: Resource
+      resource:
+        name: memory
+        targetAverageUtilization: 70
```
Comment (femilofin) [inline src/generic3/templates/horizontal-pod-autoscaler.yaml:25]: and this. we don't want to hardcode anything

File: src/generic3/values.yaml [hunk 5]
```diff
@@ -65,7 +64,16 @@ app:
       memory: 256M
     limits:
       cpu: 500m
-      memory: 512Mi
+      memory: 512M
+
+#  HPA Activation
+  replica:
+    enabled: "true"
+    min: 2
```
Comment (femilofin) [inline src/generic3/values.yaml:72]: min should be more or equal to the replica count. Can you drop the replicaCount to 2

Comment (wombolo) [reply src/generic3/values.yaml:72]: Okay, done

**Pull request merged**